borg/remote.py

```diff
@@ -316,22 +316,18 @@ def handle_error(error, res):
             if w:
                 while not self.to_send and (calls or self.preload_ids) and len(waiting_for) < 100:
                     if calls:
-                        if is_preloaded:
-                            if calls[0] in self.cache:
-                                waiting_for.append(fetch_from_cache(calls.pop(0)))
```

Comment:
I did not quite see that difference. But i think when the result is different the old code was broken. The call sequence is:
* repo.preload(chunk_ids_p1)
* repo.preload(chunk_ids_p2)
* repo.preload(chunk_ids_p3)
* repo.get_many(chunk_ids_f1, is_preload=True)
* repo.get_many(chunk_ids_f2, is_preload=True)

at the moment.
In the old code if chunk_ids_f1 contains chunk_ids that are also in chunk_ids_{p1,p2,p3} then the preloads are not yet in the cache (because that is only filled after requesting the parameters) and nothing is appended to waiting_for. I think that would result in skiping the chunk which get_many may never do because code like this:

        for id_, data in zip(ids, self.repository.get_many(ids, is_preloaded=is_preloaded)):
            yield self.key.decrypt(id_, data)

In my modification the chunks are correctly loaded, but than later are cached and then likely leaked because they are not used from the cache.